### PR TITLE
Implement mountFilter for servlet backends.

### LINKED ISSUE #472: Provide mountFilter on JettyBuilder
Filters are handy when dealing with http. You should add mountFilter on JettyBuilder, and maybe alspo a similar mechanism om the Blaze server.

### PR DESCRIPTION
**Pull request by rossabaker**
Closes #472.

As noted on #472, it's not "the http4s way," and not recommended for general use, but it is handy for those who are heavily invested in the servlet ecosystem.

### CODE REVIEW

**Comment by rossabaker** [comment]
I am not very fond of how I polluted the JettyExample and TomcatExample for what is a somewhat useful, but not particularly mainstream, feature.  Should I try to sweep this one under the rug a bit more?

File: servlet/src/main/scala/org/http4s/servlet/DefaultFilter.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+  * Mainly a convenience for our servlet examples, but, hey, why not.
+  */
+trait DefaultFilter extends Filter {
+  override def init(filterConfig: FilterConfig): Unit = {}
+
+  override def destroy(): Unit = {}
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit =
```
Comment (bryce-anderson) [inline servlet/src/main/scala/org/http4s/servlet/DefaultFilter.scala]: Should this method be final? I'm not super familiar with the semantics of using servlet filters.

**Comment by bryce-anderson** [comment]
Sorry for not getting to reviewing this sooner: I didn't actually notice it until just now.

It looks good to me although I'm not extremely experienced with the various servlet config processes. I assume you've fired the backends up and they work as expected.

I don't care one way or the other about how you've changed the examples: seems OK to me.

File: servlet/src/main/scala/org/http4s/servlet/DefaultFilter.scala (hunk 1 above)
Comment (rossabaker) [inline servlet/src/main/scala/org/http4s/servlet/DefaultFilter.scala]: It probably could be.  After more than a decade of this, I'm still not sure what implementations of ServletRequests *aren't* HttpServletRequests.

**Comment by rossabaker** [comment]
I still have misgivings about the general structure of our builders, but this fills a hole in what we have.  Going to merge if Travis likes it.

**Pull request merged**